lib/appium/helpers/wdHelper.js

```diff
@@ -0,0 +1,112 @@
+/* jshint node: true */
+/* global navigator */
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+*/
+
+'use strict';
+
+var APPIUM_SERVER_HOST = 'localhost';
+var APPIUM_SERVER_PORT = 4723;
+var WEBVIEW_WAIT_TIMEOUT = 5000;
+var IMPLICIT_WAIT_TIMEOUT = 10000;
+
+var fs = require('fs');
+var path = require('path');
+
+module.exports.getDriver = function (platform) {
+    var normalizedPlatform;
+    switch (platform.toLowerCase()) {
+        case 'android':
+            normalizedPlatform = 'Android';
+            break;
+        case 'ios':
+            normalizedPlatform = 'iOS';
+            break;
+        default:
+            throw 'Unknown platform: ' + platform;
+    }
+
+    var serverConfig = {
+        host: APPIUM_SERVER_HOST,
+        port: APPIUM_SERVER_PORT
+    };
+
+    var driverConfig = {
+        browserName: '',
+        'appium-version': '1.5',
+        platformName: normalizedPlatform,
+        platformVersion: global.PLATFORM_VERSION || '',
+        deviceName: global.DEVICE_NAME || '',
+        app: global.PACKAGE_PATH,
+        autoAcceptAlerts: true,
+    };
+
+    var driver = global.WD.promiseChainRemote(serverConfig);
+    module.exports.configureLogging(driver);
+
+    return driver
+        .init(driverConfig)
+        .setImplicitWaitTimeout(IMPLICIT_WAIT_TIMEOUT);
+};
+
+module.exports.getWD = function () {
+    return global.WD;
+};
+
+module.exports.getWebviewContext = function (driver) {
```

Comment:
What if the webview couldn't be loaded somehow? Maybe we need to limit number of attempts? 
```js
getWebviewContext = function (driver, maxAttempts) {
...
    return getWebviewContext(driver, maxAttempts - 1);
});
```